File: scala/basics/Varibles.scala
/*Use var for mutable values and val for immutable values*/
object HelloWorld {
  var myIntVar3:Int=200;
 

  
  
  /*Var keyword*/
  def test1():Unit={
    println("<<<<<<<<<<Test 1>>>>>>>>>")
     /*
     You can create mutable variable using var keyword. 
     It allows you to change value after declaration of variable.
     */ 
    /*Scala is a type infers language.You no need to specfify type explicitly*/
    var data=100;
    println(data)
    data=23;
    println(data)
  }

   /*val keyword*/
   def test2():Unit={
    println("<<<<<<<<<<Test 2>>>>>>>>>")
     /*
    You can not  resassign a varible if you define with val
     */ 
    // val data=100;
    // println(data)
    // data=23;
    // println(data)
  }
  /*var keyword and type*/
  def test3():Unit={
    println("<<<<<<<<<<Test 3:var keyword and data type>>>>>>>>>")
     /*
    You can define data type using var and :
     */ 
    var data:Int=100;
    println(data)
    
  }

  /*You can not change data type of a verible*/
  def test4():Unit={
    // println("<<<<<<<<<<Test 4:You can not change data type>>>>>>>>>")
    // var myIntVar:Int=100;
    // myIntVar="I am a string now"
  }

  /*Declaration of variable a not allowed inside function.You can do in classes*/
  def test5():Unit={
    // println("<<<<<<<<<<Test 5:Declaration of variable a not allowed inside function>>>>>>>>>")
    // var myIntVar:Int=100;
    // println(myIntVar)
    // var myIntVar2:Int;
    // myIntVar2=200;
    // println(myIntVar2)
    // //myIntVar2 will throw an error.Because you are defining and add data later
  }

  /*Global value*/
  def test6():Unit={
    println("<<<<<<<<<<Test 6:Global value>>>>>")
    println(myIntVar3)
  }



  def main(args:Array[String]) : Unit ={  
      test1();  
      test3(); 
      test5(); 
      test6(); 
   }  
}

File: golang/basics/varibles.go
package main

import "fmt"

var myGlobalVar1 int

func method1() {
	fmt.Println("<<<<<< Method1 >>>>>>>>>>.")
	//Using var keyword we can define the varibles
	//Every Go varible need to be defined with var keyword
	var myvar int
	myvar = 43
	fmt.Println(myvar)

	var myvar2 string
	myvar2 = "subbu"
	fmt.Println(myvar2)

	var myvar3 bool
	myvar3 = true
	fmt.Println(myvar3)

	var myvar4 float32
	myvar4 = 32.44
	fmt.Println(myvar4)

	var myvar5 byte
	myvar5 = 2
	fmt.Println(myvar5)

	var myvar6 rune
	myvar6 = 'a'
	fmt.Println(myvar6)

	/*You can directly add value while defining */
	var myvarwithValue int = 143
	fmt.Println(myvarwithValue)
}
func method2() {
	fmt.Println("<<<<<< Method2 >>>>>>>>>>.")
	//If we donn't init the var values,it will take the default values
	var myvar1 int
	fmt.Println(myvar1)
	var myvar2 float32
	fmt.Println(myvar2)
	var myvar3 bool
	fmt.Println(myvar3)
}

func method3() {
	fmt.Println("<<<<<< Method3 >>>>>>>>>>.")
	///You can define all varibles at once
	var i, j, k, l int
	var c, ch byte
	var f, salary float32

	fmt.Println(i + j + k + l)
	fmt.Println(c + ch)
	fmt.Println(f + salary)

}

func method4() {
	fmt.Println("<<<<<< Method4 >>>>>>>>>>.")
	//Using short variable declaration:
	//This is used inside function
	// This is a local variables which are declared and initialize in the functions
	//Not applicabulr for global varible
	//are declared by using short variable declaration.

	d := "This is defined using the Short varible"
	fmt.Println(d)

}

func method5() {
	fmt.Println("<<<<<< Method 5 >>>>>>>>>>.")
	////Global varibles
	fmt.Println(myGlobalVar1)

	//Try to define this
	//d := "This is defined using the Global varible"
	//This will give an error.As it works only for localy
}

func main() {
	method1()
	method2()
	method3()
	method4()
	method5()
}
